src/core/Transcriber.py

```diff
@@ -1,90 +1,241 @@
+    """Transcriber with true cancellation support via multiprocessing"""
+    
+    def __init__(self, ts_api, module_entry: Default.Entry, cancel_event: threading.Event):
         """
         Creates a Transcriber object to work on subtitles
-        :param ts_api: The main ts_api object to call back
-        :param module_entry: The module_entry of the job to create the
-        transcriber object for
+        
+        Args:
+            ts_api: The main ts_api object to call back
+            module_entry: The module_entry of the job to transcribe
+            cancel_event: Threading event for cancellation support
         """
         self.whisper_result = None
         self.whisper_language = None
         self.file_path: str = "./data/audioInput/" + module_entry.uid
         self.ts_api = ts_api
         self.module_entry: Default.Entry = module_entry
-
-    def start_thread(self):
-        """
-        Starts the thread that whispers the audio
-        :return: Nothing
-        """
-        whisper_thread = threading.Thread(target=self.transcriber_thread,
-                                          daemon=True)
-        whisper_thread.start()
-
-    def transcriber_thread(self):
+        self.cancel_event = cancel_event
+        self.transcription_process = None
+    
+    def transcribe(self):
         """
-        The thread to whisper an audio
-        :return: Nothing
+        Transcribe audio with true cancellation support
+        Uses multiprocessing to allow killing the transcription process
         """
         try:
-            logging.info("Starting processing for job with id "
-                         + self.module_entry.uid + "...")
-            # Whisper model
-            model_size = os.environ.get("whisper_model")
-            model = Model(model_size,
-                          models_dir="./data/models",
-                          n_threads=int(os.environ.get("whisper_cpu_threads")))
-            self.ts_api.database.change_job_entry(self.module_entry.uid,
-                                                  "whisper_model",
-                                                  model_size)
+            logging.info(f"Starting transcription for job {self.module_entry.uid}")
+            
+            # Check cancellation before starting
+            if self.cancel_event.is_set():
+                logging.info(f"Job {self.module_entry.uid} canceled before transcription")
+                self._handle_cancellation()
+                return
+            
+            # Update status to processing
+            self.ts_api.database.update_job_status(
+                self.module_entry.uid,
+                JobStatus.PROCESSING,
+                started_at=time.time()
+            )
+            
+            # Load Whisper model for language detection
+            model_size = os.environ.get("whisper_model", "tiny")
+            n_threads = int(os.environ.get("whisper_cpu_threads", 4))
```

Comment:
Do not forget to change two lines like it was in previous code for performance:
```suggestion
            model_size = os.environ.get("whisper_model")
            n_threads = int(os.environ.get("whisper_cpu_threads"))
```